lib/kredis/types.rb

```diff
@@ -72,6 +72,12 @@ def slot(key, config: :shared)
   def slots(key, available:, config: :shared)
     Slots.new configured_for(config), namespaced_key(key), available: available
   end
+
+
+  def with_callback(type, key, **options)
+    callback = options.delete(:after_change)
+    Kredis::CallbacksProxy.new(Kredis.send(type, key, **options), nil, callback)
+  end
```

Comment:
Let's make an internal factory method to help out enriching the type if needed:

```ruby
  private
    def type_from(type_klass, config, key, after_change: nil, **options)
      type_klass.new(configured_for(config), namespaced_key(key), options).then do |type|
        after_change ? Kredis::CallbacksProxy.new(type, nil, after_change) : type
    end
  end
```

Reply:
On it!